azure-kusto-data/tests/test_security.py

```diff
@@ -22,3 +22,14 @@ def test_unauthorized_exception():
         assert error.authority == "https://login.microsoftonline.com/authorityName"
         assert error.kusto_cluster == cluster
         assert error.kwargs["username"] == username
+
+def test_pickle():
```

Comment:
I can also rename this test if you like -- the real intent here is extremely short-term persistence via multiprocessing.Pool.map's pickle for moving objects to other threads.